src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -367,8 +370,27 @@ public ResultWrapper<BlockForRpc> eth_getBlockByNumber(BlockParameter blockParam
         return GetBlock(blockParameter, returnFullTransactionObjects);
     }
 
+    private bool IsBlockPruned(BlockParameter blockParameter)
+    {
+        if (_syncConfig.AncientReceiptsBarrierCalc > 1 && _syncConfig.AncientBodiesBarrierCalc > 1)
+        {
+            var lowestBlock = _syncConfig.AncientReceiptsBarrierCalc < _syncConfig.AncientBodiesBarrierCalc ? _syncConfig.AncientReceiptsBarrierCalc : _syncConfig.AncientBodiesBarrierCalc;
+            if (blockParameter.BlockNumber < lowestBlock)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
     protected virtual ResultWrapper<BlockForRpc?> GetBlock(BlockParameter blockParameter, bool returnFullTransactionObjects)
     {
+        if (IsBlockPruned(blockParameter))
+        {
+            var error = new SearchResult<Block>("Pruned history unavailable", ErrorCodes.PrunedHistoryUnavailable);
```

Comment:
Can you only do this check if the block is actually missing? Maybe the user already downloaded the block before changing the barrier.

Reply:
Good point. Addressed!